cassandra/src/main/java/org/hibernate/ogm/datastore/cassandra/CassandraDialect.java

```diff
@@ -0,0 +1,329 @@
+
+import org.hibernate.ogm.datastore.map.impl.MapTupleSnapshot;
+
+import org.hibernate.ogm.dialect.batch.spi.BatchableGridDialect;
+import org.hibernate.ogm.dialect.batch.spi.InsertOrUpdateAssociationOperation;
+import org.hibernate.ogm.dialect.batch.spi.InsertOrUpdateTupleOperation;
+import org.hibernate.ogm.dialect.batch.spi.Operation;
+import org.hibernate.ogm.dialect.batch.spi.OperationsQueue;
+import org.hibernate.ogm.dialect.batch.spi.RemoveAssociationOperation;
+import org.hibernate.ogm.dialect.batch.spi.RemoveTupleOperation;
+
+import org.hibernate.ogm.dialect.spi.AssociationContext;
+import org.hibernate.ogm.dialect.spi.AssociationTypeContext;
+import org.hibernate.ogm.dialect.spi.DuplicateInsertPreventionStrategy;
+import org.hibernate.ogm.dialect.spi.ModelConsumer;
+import org.hibernate.ogm.dialect.spi.NextValueRequest;
+import org.hibernate.ogm.dialect.spi.TupleAlreadyExistsException;
+import org.hibernate.ogm.dialect.spi.TupleContext;
+
+import org.hibernate.ogm.model.key.spi.AssociationKey;
+import org.hibernate.ogm.model.key.spi.AssociationKeyMetadata;
+import org.hibernate.ogm.model.key.spi.EntityKey;
+import org.hibernate.ogm.model.key.spi.EntityKeyMetadata;
+import org.hibernate.ogm.model.spi.Association;
+import org.hibernate.ogm.model.spi.Tuple;
+import org.hibernate.ogm.model.spi.TupleOperation;
+import org.hibernate.ogm.type.spi.GridType;
+import org.hibernate.persister.entity.Lockable;
+import org.hibernate.type.Type;
+
+import static com.datastax.driver.core.querybuilder.QueryBuilder.delete;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.eq;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.insertInto;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.select;
+import static com.datastax.driver.core.querybuilder.QueryBuilder.quote;
+
+/**
+ * Dialect implementation using CQL3 over Cassandra's native transport via java-driver.
+ *
+ * @author Jonathan Halliday
+ */
+public class CassandraDialect implements BatchableGridDialect {
+
+	private static final Log log = LoggerFactory.getLogger();
+
+	private final CassandraDatastoreProvider provider;
+	private final Session session;
+	private final ProtocolVersion protocolVersion;
+
+	public CassandraDialect(CassandraDatastoreProvider provider) {
+		this.provider = provider;
+		session = provider.getSession();
+		protocolVersion = session
+				.getCluster()
+				.getConfiguration()
+				.getProtocolOptions()
+				.getProtocolVersionEnum();
+	}
+
+	@Override
+	public LockingStrategy getLockingStrategy(Lockable lockable, LockMode lockMode) {
+		//Cassandra essentially has no workable lock strategy unless you use external tools like
+		// ZooKeeper or fancy cql overlays like waitchain.
+		return null;
+	}
+
+	private ResultSet bindAndExecute(Object[] columnValues, RegularStatement statement) {
+
+		try {
+			PreparedStatement preparedStatement = session.prepare( statement );
```

Comment:
Should the PS be cached? Some sort of cache for the current session and its statements might make sense.